api/python/t4/api.py

```diff
@@ -101,6 +102,44 @@ def delete(target):
     delete_object(bucket, path)
 
 
+def delete_package(name, registry=None):
+    if not re.match(PACKAGE_NAME_FORMAT, name):
+        raise QuiltException("Invalid package name, must contain exactly one /.")
+
+    if name not in list_packages(registry):
+        raise QuiltException("No such package exists in the given directory.")
+
+    registry_base_path = get_package_registry(fix_url(registry) if registry else None)
+    registry_url = urlparse(registry_base_path)
+    if registry_url.scheme != 'file':
+        raise NotImplementedError
+
+    pkg_namespace, pkg_subname = name.split("/")
+
+    registry_dir = pathlib.Path(parse_file_url(registry_url)).as_posix()
+    pkg_namespace_dir = pathlib.Path(registry_dir) / 'named_packages/' / pkg_namespace
+    pkg_dir = pkg_namespace_dir / pkg_subname
+    packages_path = pathlib.Path(registry_dir) / 'packages'
+
+    relevant_tophashes = []
+    for tophash_file in pkg_dir.glob('*/'):
+        # skip latest, which always duplicates a tophashed file
+        if tophash_file.name != 'latest':
+            with open(tophash_file, 'r') as f:
+                relevant_tophashes.append(f.read())
+
+        # remove the file
+        tophash_file.unlink()
+
+    pkg_dir.rmdir()
+
+    if len(list(pkg_namespace_dir.glob("*/"))) == 0:
+        pkg_namespace_dir.rmdir()
+
+    for relevant_tophash in relevant_tophashes:
+        (packages_path / relevant_tophash).unlink()
```

Comment:
Buuut... how do you know some other package doesn't have the same hash? Or more like, the same package was built twice with different names.